compiler/src/dotty/tools/dotc/transform/BetaReduce.scala

```diff
@@ -65,7 +69,8 @@ object BetaReduce:
             ref.symbol
           case _ =>
             val flags = Synthetic | (param.symbol.flags & Erased)
-            val binding = ValDef(newSymbol(ctx.owner, param.name, flags, arg.tpe.widen, coord = arg.span), arg)
+            val tpe = if arg.tpe.dealias.isInstanceOf[ConstantType] then arg.tpe.dealias else arg.tpe.widen
+            val binding = ValDef(newSymbol(ctx.owner, param.name, flags, tpe, coord = arg.span), arg).withSpan(arg.span)
```

Comment:
It's unclear to me why we cannot just use `arg.tpe.widen` as the type of the binding?

Reply:
It is to allow TermRef to definitions with constant types to be constant folded after this transformation.